packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,136 @@
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled ("ON").
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label of the Switch component.
+   */
+  label: PropTypes.node.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Parameters: 1. event 2. switch value
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  id,
+  labelId,
+  checked,
+  disabled,
+  onChange,
+  label,
+  ...customProps
+}) => {
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    sliderButtonRef.current.focus();
+    if (onChange) {
+      onChange(event, checked);
+    }
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      event.preventDefault();
+    }
+  };
+
+  const handleKeyUp = (event) => {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      sliderButtonRef.current.focus();
+      if (onChange) {
+        onChange(event, checked);
+      }
+    }
+  };
+
+  const switchClassNames = cx([
+    'switch',
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { selected: checked },
+    { unselected: !checked },
+    { 'is-disabled': disabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-disabled': disabled },
+  ]);
+
+  const statusLabelText = checked ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  return (
+    <div className={cx('switch-wrapper')}>
+      <label htmlFor={id}>
+        <div className={cx('switch-container')}>
+          <div className={cx('label-container')}>
+            <span id={labelId} className={cx('label-text')}>{label}</span>
+            <span className={cx('status-label-text')}>{statusLabelText}</span>
+          </div>
+          <div className={switchClassNames}>
+            <span className={trayClassNames}>
+              <button
+                {...customProps}
+                type="button"
+                id={id}
+                disabled={disabled}
+                aria-checked={checked}
+                aria-labelledby={labelId}
+                className={sliderClassNames}
+                role="switch"
+                tabIndex="0"
+                onClick={handleClick}
+                onKeyDown={handleOnKeyDown}
+                onKeyUp={handleKeyUp}
```

Comment:
Are the custom overrides for onKeyUp and onKeyDown necessary? Buttons natively invoke onClick for enter/space.  I commented them out `onKeyDown ` and `onKeyUp` locally and the behavior appeared to be as expected. `onClick` will be invoked for space/enter.

Reply:
I have added these events to prevent long press issue that happens with ENTER key. it is native browser behaviour that holding ENTER key triggers onClick multiple times, which causes switch to toggle between ON/OFF continuously. hence added these event handlers to prevent default behaviour of key press.

Reply:
I think I would expect a long key press to have that behavior. I wouldn't add the additional code to prevent that.

Reply:
removed [keydown and keyup events ](https://github.com/cerner/terra-core/pull/2846/commits/f4692ff31e65e7a746c1e364be845ab3210915a6#diff-ce03daf8e4362de3a438b2f322dedf6f)